src/ir/context.rs

```diff
@@ -1585,12 +1589,16 @@ impl<'ctx> BindgenContext<'ctx> {
                 match *item.kind() {
                     ItemKind::Module(..) => true,
                     ItemKind::Function(_) => {
-                        self.options().whitelisted_functions.matches(&name)
+                        cc.functions && self.options().whitelisted_functions.matches(&name)
                     }
                     ItemKind::Var(_) => {
-                        self.options().whitelisted_vars.matches(&name)
+                        cc.vars && self.options().whitelisted_vars.matches(&name)
                     }
                     ItemKind::Type(ref ty) => {
+                        if !cc.types {
+                            return false;
+                        }
```

Comment:
Woops, I forgot to remove these checks; they're unnecessary now that there is a filter for `is_enabled_for_codegen` above.